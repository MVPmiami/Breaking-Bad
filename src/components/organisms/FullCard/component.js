import styles from "./style.module.scss";
import ErrorMessage from "../../atoms/Error";
import Loader from "../../atoms/Loader";

const FullCard = ({ person, isLoader, isError, randomQuote }) => {
  return (
    <>
      {isError ? (
        <ErrorMessage messege="Person not found" />
      ) : isLoader ? (
        <Loader />
      ) : (
        <div className={styles.wrapper}>
          <div className={styles.photoWrapper}>
            <img
              className={styles.photo}
              src={person[0].img}
              alt={person[0].name}
            ></img>
          </div>
          <div className={styles.info}>
            <div className={styles.status}>{person[0].status}</div>
            <div className={styles.name}>{person[0].name}</div>
            <div className={styles.birthday}>
              <span className={styles.lightText}>Date of birth: </span>
              {person[0].birthday}
            </div>
            <div className={styles.nickName}>
              <span className={styles.lightText}>Nickname: </span>
              {person[0].nickname}
            </div>
            <div className={styles.quote}>
              <span className={styles.lightText}>Quote: </span>
              {randomQuote}
            </div>
          </div>
        </div>
      )}
    </>
  );
};

export default FullCard;
