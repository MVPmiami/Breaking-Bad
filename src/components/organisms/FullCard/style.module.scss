@import "./../../../assets/styles/style.scss";

.wrapper {
  width: 74.563rem;
  display: flex;
  .photoWrapper {
    max-width: 36.5rem;
    height: 36.563rem;
    object-fit: cover;
    margin-right: 6.4375rem;
    overflow-y: hidden;
    .photo {
      width: 100%;
      height: auto;
    }
  }
  .info {
    font-family: "Roboto";
    .status {
      @include status($statusText, $statusBcg);
      max-width: 92px;
      margin-bottom: 2.5rem;
    }
    .name {
      font-style: normal;
      font-weight: 500;
      font-size: 2rem;
      line-height: 2.5rem;
      margin-bottom: 1rem;
    }
    .birthday,
    .nickName,
    .quote {
      color: $dataText;
      margin-bottom: 1rem;
    }
    .lightText {
      color: $dataText;
      opacity: 0.4;
    }
    .quote {
      margin-top: 2rem;
    }
  }
}
@media (max-width: 1240px) {
  .wrapper {
    width: 50rem;
    display: flex;
    .photoWrapper {
      min-width: 26.5rem;
      height: 26.563rem;
      object-fit: cover;
      margin-right: 4.4375rem;
      overflow-y: hidden;
      .photo {
        width: 100%;
        height: 135%;
      }
    }
  }
}
@media (max-width: 820px) {
  .wrapper {
    width: 30rem;
    flex-direction: column;
    .photoWrapper {
      min-width: 18.5rem;
      height: 18.563rem;
      margin-right: 4.4375rem;
    }
    .status {
      margin: 1rem 0;
    }
    .quote {
      width: 25.5rem;
    }
  }
}
@media (max-width: 490px) {
  .wrapper {
    width: 20rem;
    flex-direction: column;
    align-items: center;
    .photoWrapper {
      max-width: 5.5rem;
      height: 19.563rem;
      margin-right: 0;
    }
    .info {
      width: 18.5rem;
      .status {
        margin: 1rem 0;
      }
      .quote {
        width: 18.5rem;
      }
    }
  }
}
