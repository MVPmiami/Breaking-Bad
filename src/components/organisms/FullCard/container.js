import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useParams } from "react-router-dom";
import FullCard from "./component";
import { loadPersonById } from "./../../../store/actions/persons";

export const PersonFullCardContainer = () => {
  const dispatch = useDispatch();
  const { id } = useParams();
  const person = useSelector((state) => state.cardListReducer.currentPerson);
  const isError = useSelector(
    (state) => state.cardListReducer.isErrorCurrentPerson
  );
  const isLoader = useSelector(
    (state) => state.cardListReducer.isLoadCurrentPerson
  );

  useEffect(() => {
    dispatch(loadPersonById(id));
  }, [dispatch]);

  return <FullCard person={person} isError={isError} isLoader={isLoader} />;
};

export const container = PersonFullCardContainer;
