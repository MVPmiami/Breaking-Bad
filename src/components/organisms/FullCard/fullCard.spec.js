import React from "react";
import FullCard from "../../../../src/components/organisms/FullCard/component";

const setUp = (props) => shallow(<FullCard {...props} />);

describe("should render FullCard component", () => {
  let component;
  let person = [
    {
      char_id: 1,
      name: "Walter White",
      birthday: "09-07-1958",
      occupation: ["High School Chemistry Teacher", "Meth King Pin"],
      img: "https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg",
      status: "Presumed dead",
      nickname: "Heisenberg",
      appearance: [1, 2, 3, 4, 5],
      portrayed: "Bryan Cranston",
      category: "Breaking Bad",
      better_call_saul_appearance: [],
    },
  ];
  let isLoader = false;
  let isError = false;
  let randomQuote = "test";
  beforeEach(() => {
    component = setUp({ person, isLoader, isError, randomQuote });
  });

  it("should contain .wrapper ", () => {
    expect(component).toMatchSnapshot();
  });
});

describe("should render FullCard component with ErrorMessage", () => {
  let component;
  let isError = true;
  beforeEach(() => {
    component = setUp({ isError });
  });

  it("should contain ErrorMessage ", () => {
    expect(component).toMatchSnapshot();
  });
});

describe("should render FullCard component with Loader", () => {
  let component;
  let isError = false;
  let isLoader = true;
  let randomQuote = "test";
  beforeEach(() => {
    component = setUp({ isError, isLoader, randomQuote });
  });

  it("should contain Loader ", () => {
    expect(component).toMatchSnapshot();
  });
});

describe("should render FullCard component with randomQuote", () => {
  let component;
  let isError = false;
  let isLoader = false;
  let person = [
    {
      char_id: 1,
      name: "Walter White",
      birthday: "09-07-1958",
      occupation: ["High School Chemistry Teacher", "Meth King Pin"],
      img: "https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg",
      status: "Presumed dead",
      nickname: "Heisenberg",
      appearance: [1, 2, 3, 4, 5],
      portrayed: "Bryan Cranston",
      category: "Breaking Bad",
      better_call_saul_appearance: [],
    },
  ];
  let randomQuote = "test";
  beforeEach(() => {
    component = setUp({ person, isLoader, isError, randomQuote });
  });

  it("should contain randomQuote ", () => {
    expect(component).toMatchSnapshot();
  });
});
