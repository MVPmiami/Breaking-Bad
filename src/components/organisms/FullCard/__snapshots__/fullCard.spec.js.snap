// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FullCard component should contain .wrapper  1`] = `
<Fragment>
  <div
    className="wrapper"
  >
    <div
      className="photoWrapper"
    >
      <img
        alt="Walter White"
        className="photo"
        src="https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg"
      />
    </div>
    <div
      className="info"
    >
      <div
        className="status"
      >
        Presumed dead
      </div>
      <div
        className="name"
      >
        Walter White
      </div>
      <div
        className="birthday"
      >
        <span
          className="lightText"
        >
          Date of birth: 
        </span>
        09-07-1958
      </div>
      <div
        className="nickName"
      >
        <span
          className="lightText"
        >
          Nickname: 
        </span>
        Heisenberg
      </div>
      <div
        className="quote"
      >
        <span
          className="lightText"
        >
          Quote: 
        </span>
        test
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`should render FullCard component with ErrorMessage should contain ErrorMessage  1`] = `
<Fragment>
  <ErrorMessage
    messege="Person not found"
  />
</Fragment>
`;

exports[`should render FullCard component with Loader should contain Loader  1`] = `
<Fragment>
  <Loader />
</Fragment>
`;

exports[`should render FullCard component with randomQuote should contain randomQuote  1`] = `
<Fragment>
  <div
    className="wrapper"
  >
    <div
      className="photoWrapper"
    >
      <img
        alt="Walter White"
        className="photo"
        src="https://images.amcnetworks.com/amc.com/wp-content/uploads/2015/04/cast_bb_700x1000_walter-white-lg.jpg"
      />
    </div>
    <div
      className="info"
    >
      <div
        className="status"
      >
        Presumed dead
      </div>
      <div
        className="name"
      >
        Walter White
      </div>
      <div
        className="birthday"
      >
        <span
          className="lightText"
        >
          Date of birth: 
        </span>
        09-07-1958
      </div>
      <div
        className="nickName"
      >
        <span
          className="lightText"
        >
          Nickname: 
        </span>
        Heisenberg
      </div>
      <div
        className="quote"
      >
        <span
          className="lightText"
        >
          Quote: 
        </span>
        test
      </div>
    </div>
  </div>
</Fragment>
`;
